### PR TITLE
refactor(components/loading-spinner): use typescript

### PR DESCRIPTION
**Pull request by adnasa**
#### Summary

- migrate `LoadingSpinner` to typescript

### CODE REVIEW

**Review by tdeekens — APPROVED** [review]

File: packages/components/loading-spinner/src/loading-spinner.tsx [hunk 1]
```diff
@@ -24,10 +23,25 @@ const sizePerScale = {
...
 
-const LoadingSpinner = (props) => {
+type TLoadingSpinnerProps = {
+  /**
+   * Set the size of the loading spinner.
+   */
+  scale: 's' | 'l';
+  /**
+   * -
+   */
```
Comment (emmenko) [inline packages/components/loading-spinner/src/loading-spinner.tsx:36]: Let's provide a more meaningful description

**Review by mohib0306 — APPROVED** [review]
👍

**Comment by adnasa** [comment]
ugh. I see now the issue.
will be addressed

**Comment by tdeekens** [comment]
Rebased over `main` cause I want to see if CI and VRTs are happy now.

**Comment by adnasa** [comment]
thanks for this <3

**Comment by tdeekens** [comment]
Hm. There is an inherent problem with this branch and current state. Something is wrong that is not CI :)

File: packages/components/loading-spinner/src/loading-spinner.tsx [hunk 2]
```diff
@@ -24,10 +23,27 @@ const sizePerScale = {
...
-const LoadingSpinner = (props) => {
+type TLoadingSpinnerProps = {
+  /**
+   * Set the size of the loading spinner.
+   */
+  scale: 's' | 'l';
+  /**
+   * The content rendered inside the `LoadingSpinner`.
+   * <br />
+   * It will be rendered as `Text.Detail` with `secondary` tone.
```
Comment (emmenko) [inline packages/components/loading-spinner/src/loading-spinner.tsx:37]: Nit: I think we can omit this info

**Review by emmenko — APPROVED** [review]

**Comment by adnasa** [comment]
oh well.

**Pull request merged**